### PR TITLE
docs: Build and release

### PR DESCRIPTION
**Pull request by maxwellmattryan**
# Description of change

- Create release guide (look in Engineering section of Confluence - [link](https://iotafoundation.atlassian.net/wiki/spaces/IF/pages/1834549249/Firefly))
- Create [`mobile/README.md`](https://github.com/iotaledger/firefly/blob/docs/build-and-release/packages/mobile/README.md) and [`shared/README.md`](https://github.com/iotaledger/firefly/blob/docs/build-and-release/packages/shared/README.md)
- Modify [`desktop/README.md`](https://github.com/iotaledger/firefly/blob/docs/build-and-release/packages/desktop/README.md) and [`backend/README.md`](https://github.com/iotaledger/firefly/blob/docs/build-and-release/packages/backend/README.md)
    - Improve steps for Windows setup (shoutout to @begonaalvarezd and @Thoralf-M for troubleshooting Windows - we made it to the other side!)
- Modify [`README.md`](https://github.com/iotaledger/firefly/blob/docs/build-and-release/README.md) in root dir

## Links to any relevant issues

None

## Type of change

- Docs (changes to the documentation)

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/README.md [hunk 1]
```diff
@@ -1,74 +1,113 @@
...
+```bash
 # in packages/desktop
-yarn compile:win
+yarn compile:<platform>
 ```
-Change win to your operating system - mac, win or linux.
 
-Mac users need to `return true` in the notarization script in `packages/desktop/notarization`.
+Change `<platform>` as necessary (`win`, `mac`, and `linux`). 
+MacOS users __must__ `return true` in the notarization script in `packages/desktop/notarization`.
```
Comment (rajivshah3) [inline packages/desktop/README.md:99]: To make it a bit easier (#1414):

```suggestion
MacOS users __must__ set an environment variable in order to skip notarization (`MACOS_SKIP_NOTARIZATION=true`).
```

File: packages/shared/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Contains all of the business logic for the different features
+in Firefly (e.g. handling migrations, transferring funds). Wrapper 
+functions for the API (wallet.rs) or functions that use those are 
+defined here.
+
+## `locales`
+
+Contains all of the data for translations in other languages 
+(`en.json` (English) is used as a base for the translation).
```
Comment (rajivshah3) [inline packages/shared/README.md:18]:
```suggestion
(`en.json` (English) is used as a base for the translation).
Please do not modify any files besides `en.json` as the others are synced from [Crowdin](https://crowdin.com/project/iota-firefly).
```

File: packages/desktop/README.md [hunk 3]
```diff
@@ -1,74 +1,113 @@
...
+## Setup
 
-For all platforms, the following are required:
+The following __must__ be installed on all platforms:
 
--   [Node.js](https://nodejs.org/en/) 12+ (note: there may be issues with Node.js 15 on Windows)
--   [Yarn](https://classic.yarnpkg.com/en/docs/install)
--   [Rust](https://www.rust-lang.org/tools/install)
+- [Node.js](https://nodejs.org/en/) 12+ (NOTE: There may be issues with Node.js 15 on Windows)
+- [Yarn](https://classic.yarnpkg.com/en/docs/install) (NOTE: `npm` is required for Windows setup)
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/desktop/README.md:10]: Yup 😅

**Review by rajivshah3 — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -94,7 +94,7 @@ Otherwise, your feature may not be approved at all.
 
 ### Building a new feature
 
-To build a new feature, check out a new branch based on the `master` branch, and be sure to document any public-facing APIs, using Rust code comments.
+To build a new feature, check out a new branch based on the `develop` branch, and be sure to document any public-facing APIs, using Rust code comments.
```
Comment (rajivshah3) [inline .github/CONTRIBUTING.md:97]: Nice catch!

**Review by rajivshah3 — APPROVED** [review]

**Pull request merged**